docs/frontend-system/building-plugins/migrating.md

```diff
@@ -0,0 +1,125 @@
+---
+id: migrating
+title: Migrating an existing Frontend Plugin to the New Frontend System
+sidebar_label: Migration Guide
+# prettier-ignore
+description: How to migrate an existing frontend plugin to the new frontend system
+---
+
+This guide allows you to migrate a frontend plugin and its own components, routes, apis to the new frontend system.
+
+The main concept is that routes, components, apis are now extensions. You can use the appropriate extension creators to migrate all of them to extensions.
+
+## Pages
```

Comment:
I think wrapping these in an extension migration section could be good as there's gonna need to be a fair bit more content. In particular we'll likely want to list how to migrate existing patterns to new extension creators, like entity page content, scaffolder field extensions, etc.

With that it might also make sense to start by covering how to migrate the other `createPlugin` options first, just to get that out of the way? In particular the routes and how to move over APIs.

Reply:
yeah I've used this pattern because a frontend plugin without extensions is a dead plugin 😅 but happy to change